mythril/mythril/mythril_config.py

```diff
@@ -0,0 +1,220 @@
+            "",
+        )
+        config.set(
+            "defaults", "#– To connect to local host use dynamic_loading: localhost", ""
+        )
+        config.set("defaults", "dynamic_loading", "infura")
+
+    def set_api_leveldb(self, leveldb_path: str) -> None:
+        """
+        """
+        self.eth_db = EthLevelDB(leveldb_path)
+
+    def set_api_rpc_infura(self) -> None:
+        """Set the RPC mode to INFURA on Mainnet."""
+        log.info("Using INFURA Main Net for RPC queries")
+        self.eth = EthJsonRpc("mainnet.infura.io", 443, True)
+
+    def set_api_rpc(self, rpc: str = None, rpctls: bool = False) -> None:
+        """
+        Sets the RPC mode to either of ganache or infura
+        :param rpc: either of the strings - ganache, infura-mainnet, infura-rinkeby, infura-kovan, infura-ropsten
+        """
+        if rpc == "ganache":
+            rpcconfig = ("localhost", 8545, False)
+        else:
+            m = re.match(r"infura-(.*)", rpc)
+            if m and m.group(1) in ["mainnet", "rinkeby", "kovan", "ropsten"]:
+                rpcconfig = (m.group(1) + ".infura.io", 443, True)
+            else:
+                try:
+                    host, port = rpc.split(":")
+                    rpcconfig = (host, int(port), rpctls)
+                except ValueError:
+                    raise CriticalError(
+                        "Invalid RPC argument, use 'ganache', 'infura-[network]' or 'HOST:PORT'"
+                    )
+
+        if rpcconfig:
+            log.info("Using RPC settings: %s" % str(rpcconfig))
+            self.eth = EthJsonRpc(rpcconfig[0], int(rpcconfig[1]), rpcconfig[2])
+        else:
+            raise CriticalError("Invalid RPC settings, check help for details.")
+
+    def set_api_rpc_localhost(self) -> None:
+        """Set the RPC mode to a local instance."""
+        log.info("Using default RPC settings: http://localhost:8545")
+        self.eth = EthJsonRpc("localhost", 8545)
+
+    def set_api_from_config_path(self) -> None:
+        """Set the RPC mode based on a given config file."""
+        config = ConfigParser(allow_no_value=False)
+        # TODO: Remove this after this issue https://github.com/python/mypy/issues/2427 is closed
+        config.optionxform = str  # type:ignore
+        config.read(self.config_path, "utf-8")
+        if config.has_option("defaults", "dynamic_loading"):
+            dynamic_loading = config.get("defaults", "dynamic_loading")
+        else:
+            dynamic_loading = "infura"
+        self._set_rpc(dynamic_loading)
+
+    def _set_rpc(self, rpc_type: str) -> None:
+        """
+        Sets rpc based on the type
+        :param rpc_type: The type of connection: like infura, ganache, localhost
+        :return:
+        """
+        if rpc_type == "infura":
+            self.set_api_rpc_infura()
+        elif rpc_type == "localhost":
+            self.set_api_rpc_localhost()
+        else:
+            self.set_api_rpc(rpc_type)
```

Comment:
Is there a reason `rpc_type` is checked for `infura` and `localhost` in `_set_rpc()`, but checked for `ganache`, `infura-*`, and `host:port` in `set_api_rpc()`?

Reply:
set_api_rpc is directly used when the --rpc flag is given. so in the _set_rpc() case it is just reusing it.

Reply:
Got it.